packages/driver/src/cy/commands/actions/attachFile.ts

```diff
@@ -0,0 +1,195 @@
+import _ from 'lodash'
+import Promise from 'bluebird'
+
+import $dom from '../../../dom'
+import $errUtils from '../../../cypress/error_utils'
+import $actionability from '../../actionability'
+import { addEventCoords, dispatch } from './trigger'
+
+// TODO: Does not currently need to use promises, but will once we allow users to
+// load fixtures and read files. Writing it async from the start to avoid
+// needing to change it later.
+/*
+ * Turns a user-provided file - a string shorthand, Buffer, or object
+ * into an object of form {
+ *   contents,
+ *   fileName,
+ *   lastModified,
+ *   mimeType?,
+ * }.
+ *
+ * Or throws an error. Users do many strange things, and this is where
+ * we warn them and suggest how to fix it.
+ */
+const parseFile = (options) => {
+  return (file: any, index: number, filesArray: any[]): Promise<Cypress.FileReferenceObject> => {
+    if (typeof file === 'string') {
+      throw new Error('Support for string shorthands is still TODO')
+    }
+
+    if (Buffer.isBuffer(file)) {
+      file = { contents: file }
+    }
+
+    if (!file || file.contents == null) {
+    // Different error messages if the user passed in one file vs. an array of files
+      if (filesArray.length > 1) {
+        $errUtils.throwErrByPath('attachFile.invalid_array_file_reference', {
+          onFail: options._log,
+          args: { file: JSON.stringify(file), index },
+        })
+      } else {
+        $errUtils.throwErrByPath('attachFile.invalid_single_file_reference', {
+          onFail: options._log,
+          args: { file: JSON.stringify(file) },
+        })
+      }
+    }
+
+    if (!_.isString(file.contents) && !Buffer.isBuffer(file.contents)) {
+      file.contents = JSON.stringify(file.contents)
+    }
+
+    return Promise.resolve(_.defaults({}, file, {
```

Comment:
Ah, neat, I'm happy to use async/await. Just figured that since we exposed Bluebird promises to the world we'd want to use them internally as well. Will change.